### PR TITLE
[OPS, TEST] Add onnx test for batched_nms

### PR DESCRIPTION
**Pull request by NicolasHug**
Towards debugging the issues in https://github.com/pytorch/vision/pull/3426

This PR just adds a minor simple onnx test for `batch_nms`, as there is already one for `nms`.

### CODE REVIEW

**Comment by fmassa** [comment]
@neginraoof can you have a look? The test seems to be failing for `batched_nms`, although it's supposed to be an operator that already works for ONNX.

**Comment by neginraoof** [comment]
I'm going to look into the export issue.

**Review by jiafatom — COMMENTED** [review]
Can you also have a PR to add this test in pytorch repo [here](https://github.com/pytorch/pytorch/blob/master/test/onnx/test_pytorch_onnx_onnxruntime.py)? we want to keep pytorch and pytorch/vision onnx tests in sync. Thank you.

**Comment by neginraoof** [comment]
Looks like this is an existing issue (not related to the update) with higher number of boxes in NMS.
The test works with num_boxes=5. Will look into fixing this.
Can you also add a test for num_boxes > 4000 ?

**Comment by NicolasHug** [comment]
> Can you also have a PR to add this test in pytorch repo

> Would you please also add a test for num_boxes > 4000 ?

done :)

File: test/test_onnx.py [hunk 1]
```diff
@@ -92,6 +92,18 @@ def forward(self, boxes, scores):
...
         self.run_model(Module(), [(boxes, scores)])
 
+    def test_batched_nms(self):
+        class Module(torch.nn.Module):
+            def forward(self, boxes, scores, idxs):
+                return ops.batched_nms(boxes, scores, idxs, 0.5)
+
+        for num_boxes in (5, 100, 5_000):
+            boxes = torch.rand(num_boxes, 4)
+            boxes[:, 2:] += torch.rand(num_boxes, 2)
```
Comment (BowenBao) [inline test/test_onnx.py:102]:
```suggestion
            boxes[:, 2:] += boxes[:, :2]
```

Looks like the root cause for mismatch is that for many cases, the random box is not having x2 > x1 && y2 > y1. This leads to undefined behavior, and thus mismatch. I did a quick check with above change, and results are all matching. (although the above change is not entirely legal as well since it could potentially have x2,y2 > 1, but that help prove the assumption)

Comment (NicolasHug) [reply test/test_onnx.py]: Thanks @BowenBao , you are correct, the assumptions of `nms` for the dimensions weren't respected in the test. In fact `test_nms` was incorrect too so I fixed it as well.

>  it could potentially have x2,y2 > 1

I think that's fine, there's no upper limit on x2 and y2 - unless I'm missing something here?

File: test/test_onnx.py [hunk 2]
```diff
@@ -82,16 +82,29 @@ def to_numpy(tensor):
                     raise
 
     def test_nms(self):
-        boxes = torch.rand(5, 4)
-        boxes[:, 2:] += torch.rand(5, 2)
-        scores = torch.randn(5)
+        num_boxes = 100
+        boxes = torch.rand(num_boxes, 4)
+        boxes[:, 2:] += 1 + torch.rand(num_boxes, 2)
```
Comment (fmassa) [inline test/test_onnx.py:87]: for a better fix, can you do as @BowenBao proposed and do
```python
boxes[:, 2:] += boxes[:, :2]
```
?
This way we ensure that the boxes are always well-formed (with potentially some boxes being 0, but we can add an eps to avoid that). The current fix works because `rand` is between 0-1, but if we were to have a different distribution we would have similar problems (and I was bitten by that in the past)

**Review by fmassa — APPROVED** [review]
Thanks!

File: test/test_onnx.py (hunk 1 above)
Comment (BowenBao) [reply test/test_onnx.py]: Thanks for your fix. Yes after checking the doc again i think it's fine. I thought for a moment the coordinates are required to be normalized to be within [0,1].
